draft-ietf-masque-h3-datagram.md

```diff
@@ -455,6 +442,74 @@ it is allowed to send an HTTP/3 datagram and how to process them from
 capsule.
 
 
+# Context Extensibility {#context-ext}
+
+In order to facilitate extensibility of contexts, the
+REGISTER_DATAGRAM_CONTEXT, REGISTER_DATAGRAM_NO_CONTEXT, and the
+CLOSE_DATAGRAM_CONTEXT capsules carry a Context Extensions field. That
+field contains a sequence of context extensions:
+
+~~~
+  Context Extensions {
+    Context Extension (..) ...,
+  }
+~~~
+
+Each context extension is encoded as a (type, length, value) tuple:
+
+~~~
+  Context Extension {
+    Context Extension Type (i),
+    Context Extension Length (i),
+    Context Extension Value (..),
+  }
+~~~
+
+Context Extension Types are registered with IANA, see {{iana-ext-types}}. The
+Context Extension Length field contains the length of the Context Extension
+Value field in bytes. The semantics of the Context Extension Value field are
+defined by the corresponding Context Extension Type.
+
+
+## The CLOSE_CODE Context Extension Type {#close-code}
+
+The CLOSE_CODE context extension type (type=0x00) allows an endpoint to provide
+additional information as to why a datagram context was closed. This type SHALL
+only be sent in CLOSE_DATAGRAM_CONTEXT capsules. Its Context Extension Value
+field consists of a single variable-length integer which contains the close
+code. The following codes are defined:
+
+NO_ERROR (code=0x00):
+
+: This indicates that the registration was closed without any additional
+information.
+
+DENIED (code=0x01):
+
+: This indicates that the sender has rejected the context registration based on
+its local policy. The endpoint that had originally registered this context MUST
+NOT try to register another context with the same context extensions on this
+stream.
+
+RESSOURCE_LIMIT (code=0x02):
+
+: This indicates that the context was closed to save resources. The recipient
+SHOULD limit its future registration of resource-incentive contexts.
+
+If CLOSE_CODE is not present in a CLOSE_DATAGRAM_CONTEXT capsule, the recipient
+SHALL treat it as if the NO_ERROR code was present. Close codes are registered
```

Comment:
Omitting the code seems like a micro optimization. Is there a strong case for this?

Reply:
That's a good point, fixed.